zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/changes/PartitionDeleteExporterApplier.java

```diff
@@ -0,0 +1,109 @@
+
+final class PartitionDeleteExporterApplier implements MemberOperationApplier {
+
+  private final int partitionId;
+  private final MemberId memberId;
+  private final String exporterId;
+  private final PartitionChangeExecutor partitionChangeExecutor;
+
+  public PartitionDeleteExporterApplier(
+      final int partitionId,
+      final MemberId memberId,
+      final String exporterId,
+      final PartitionChangeExecutor partitionChangeExecutor) {
+    this.partitionId = partitionId;
+    this.memberId = memberId;
+    this.exporterId = exporterId;
+    this.partitionChangeExecutor = partitionChangeExecutor;
+  }
+
+  @Override
+  public MemberId memberId() {
+    return memberId;
+  }
+
+  @Override
+  public Either<Exception, UnaryOperator<MemberState>> initMemberState(
+      final ClusterConfiguration currentClusterConfiguration) {
+
+    if (!currentClusterConfiguration.hasMember(memberId)) {
+      return Either.left(
+          new IllegalStateException(
+              String.format(
+                  "Expected to delete exporter, but the member '%s' does not exist in the cluster",
+                  memberId)));
+    }
+
+    final MemberState member = currentClusterConfiguration.getMember(memberId);
+    final var partitionExistsInLocalMember = member.hasPartition(partitionId);
+
+    if (!partitionExistsInLocalMember) {
+      return Either.left(
+          new IllegalStateException(
+              String.format(
+                  "Expected to delete exporter, but the member '%s' does not have the partition '%s'",
+                  memberId, partitionId)));
+    }
+
+    final var partitionHasExporter =
+        member.getPartition(partitionId).config().exporting().exporters().containsKey(exporterId);
+
+    if (!partitionHasExporter) {
+      return Either.left(
+          new IllegalStateException(
+              String.format(
+                  "Expected to delete exporter, but the partition '%s' does not have the exporter '%s'",
+                  partitionId, exporterId)));
+    }
+
+    // No need to change the state
+    return Either.right(memberstate -> memberstate);
+  }
+
+  @Override
+  public ActorFuture<UnaryOperator<MemberState>> applyOperation() {
+    final var result = new CompletableActorFuture<UnaryOperator<MemberState>>();
+
+    partitionChangeExecutor
+        .deleteExporter(partitionId, exporterId)
+        .onComplete(
+            (nothing, error) -> {
+              if (error == null) {
+                result.complete(
+                    memberState ->
+                        memberState.updatePartition(
+                            partitionId,
+                            partition ->
+                                partition.updateConfig(
+                                    config -> deleteExporter(config, exporterId))));
+              } else {
+                result.completeExceptionally(error);
+              }
+            });
+
+    return result;
```

Comment:
:wrench: You can simplify is using `ActorFuture#thenApply`
```suggestion
 return partitionChangeExecutor
        .deleteExporter(partitionId, exporterId)
        .thenApply(
            (nothing) -> {
              return memberState ->
                  memberState.updatePartition(
                      partitionId,
                      partition ->
                          partition.updateConfig(config -> deleteExporter(config, exporterId)));
            });
```